Dockerfile

```diff
@@ -0,0 +1,30 @@
+# syntax=docker/dockerfile:1
+
+# Build stage
+FROM rust:1.68 as builder
+
+RUN <<EOF
+    set -e
+
+    apt-get update
+    apt-get install -y build-essential
+
+    rm -rf /var/lib/apt/lists/*
+EOF
+
+WORKDIR /app
+
+COPY . .
+
+RUN --mount=type=cache,target=$RUSTUP_HOME,from=rust,source=$RUSTUP_HOME \
+    --mount=type=cache,target=$CARGO_HOME,from=rust,source=$CARGO_HOME \
+    --mount=type=cache,target=target \
+    cargo install --root output --path .
+
+
+# Main stage
+FROM debian:bullseye-slim
```

Comment:
Why debian here?

Reply:
what should be there? alpine? Debian is a standard choice for Rust

Reply:
Debian is fine for now, we can change it in the future.